protected/humhub/modules/admin/controllers/InformationController.php

```diff
@@ -78,15 +83,42 @@ public function actionAbout()
 
     public function actionPrerequisites()
     {
-        return $this->render('prerequisites', ['checks' => \humhub\libs\SelfTest::getResults()]);
+        return $this->render('prerequisites', ['checks' => SelfTest::getResults()]);
     }
 
-    public function actionDatabase()
+    public function actionDatabase($migrations = self::DB_ACTION_CHECK)
     {
+
+        $request = Yii::$app->request;
+
+        if ($migrations === self::DB_ACTION_RUN) {
+            $migrationOutput = sprintf(
+                "%s\n%s",
+                MigrateController::webMigrateAll(),
+                SettingController::flushCache()
+            );
+
+            $linkRunMigrations = [
+                'caption' => Yii::t('AdminModule.information', 'Update migration information'),
```

Comment:
Do we really need to have different captions/buttons/action here?

I see only two situations:

- There are new migrations to execute. --> Show button
- There are no migration. --> No button

Reply:
There is a semi-third state:

-  migrations are run and the result displayed in the window. So the "Update migration information" is show to re-check. would be the same as reloading the page.

Is kinda verifying that everything is fine now. Or after a migration failed to see what is now actually pending.

Reply:
> > But then, why not simply add a "Reload" button to the view on condition `!empty($migrationOutput)`.
> 
> Actually, `$migrationOutput` will never be empty, as "empty" is when it says "no migrations". But I get your point.
> 
> I'm not quite sure, why "reload" would be more indicative than "Update migration information". I find it a bit ambiguois, as it could also mean "reload the migrations".
> 
> How about "refresh"?

"Refresh" would be also fine for me. We could also do something like `str_replace('No new migrations!', '')` on controller level. And on view: 

```
<?php if (empty($migrationOutput)): ?>

Some more user friendly and translated message, that the database is up 2 date.

<?php endif; ?> 
```

> 
> > Why do we need `DB_ACTION_CHECK` constants and controller logic? Are these perhaps leftovers from the Ajax solution?
> 
> I have used constants, as I did not want to just use free-text strings. But now that we have abolished PJAX (which needed to distinguish three states: a) primary load, b) migration check, and c) migration run) we do only have to distinguish two states (load with check vs. load with run). So we could do it similar to the refresh search thingy and just use `1`, if you prefer.
> 
> > controller logic
> 
> We need that anyway to distinguish the two requests and run different migration commands (`up` vs `new`)

Please see the view file below, for an idea, of controller logic. Did I missed something?